python/pyspark/ml/base.py

```diff
@@ -59,6 +59,12 @@ def fit(self, dataset, params=None):
             return [self.fit(dataset, paramMap) for paramMap in params]
         elif isinstance(params, dict):
             if params:
+                if type(params.keys()[0]) is str:
```

Comment:
Ah, I meant, `isinstance(params.keys()[0],str)`. Direct type comparison is a strict type comparison which does not take care of subclasses. I believe it is a common pattern to check the types. For example,

```python
>>> class A(str):
...     pass
...
>>> isinstance(A(), str)
True
>>> type(A()) is str
False
```

This might be major. You can imagine giving an argument as `OrderedDict` which extends `dict`.

Another reason is consistency across codebase. I believe `isinstance` is more widely used in the codes.